cpp/examples/tpch/q1.cpp

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/ast/expressions.hpp>
+#include <cudf/column/column.hpp>
+#include <cudf/scalar/scalar.hpp>
+
+#include "utils.hpp"
+
+/*
+create view lineitem as select * from '~/tpch_sf1/lineitem/part-0.parquet';
+
+select
+    l_returnflag,
+    l_linestatus,
+    sum(l_quantity) as sum_qty,
+    sum(l_extendedprice) as sum_base_price,
+    sum(l_extendedprice * (1 - l_discount)) as sum_disc_price,
+    sum(l_extendedprice * (1 - l_discount) * (1 + l_tax)) as sum_charge,
+    avg(l_quantity) as avg_qty,
+    avg(l_extendedprice) as avg_price,
+    avg(l_discount) as avg_disc,
+    count(*) as count_order
+from
+    lineitem
+where
+        l_shipdate <= date '1998-09-02'
+group by
+    l_returnflag,
+    l_linestatus
+order by
+    l_returnflag,
+    l_linestatus;
+*/
+
+std::unique_ptr<cudf::column> calc_disc_price(std::unique_ptr<table_with_cols>& table) {
+    auto one = cudf::fixed_point_scalar<numeric::decimal64>(1);
+    auto discount = table->column("l_discount");
+    auto one_minus_discount = cudf::binary_operation(one, discount, cudf::binary_operator::SUB, discount.type());
+    auto extended_price = table->column("l_extendedprice");
+    auto disc_price_type = cudf::data_type{cudf::type_id::DECIMAL64, -4};
+    auto disc_price = cudf::binary_operation(extended_price, one_minus_discount->view(), cudf::binary_operator::MUL, disc_price_type);
+    return disc_price;
+}
+
+std::unique_ptr<cudf::column> calc_charge(std::unique_ptr<table_with_cols>& table, std::unique_ptr<cudf::column>& disc_price) {
```

Comment:
Can this take a `column_view` instead of a `unique_ptr` reference? Same question for the table, which could be a `table_view`, but I don't know if we have a "view" type with named columns like `table_with_cols`.

Maybe the better solution is to accept `column_view` for all the parameters of the calculation functions, and move the name resolution logic like `table->column("l_tax")` into the calling code in `main`.